src/Http.elm

```diff
@@ -443,8 +443,12 @@ handleResponse handle response =
             fail (UnexpectedPayload "Response body is a blob, expecting a string.")
 
   else
-
-      fail (BadResponse response.status response.statusText)
+      let
+        text = case response.value of
+          Text str -> str
+          _ -> response.statusText
```

Comment:
`statusText` is the most useless of all these values, even in case of an empty body.

I'm not sure how receiving `BadResponse 422 "ERROR"` is better than `BadResponse 422 ""`.

However, this patch is about the non-empty, text response (for example JSON) with non 200 status code.